gc/base/SweepHeapSectioning.cpp

```diff
@@ -458,3 +458,25 @@ MM_SweepHeapSectioning::reassignChunks(MM_EnvironmentBase *env)
 
 	return totalChunkCount;
 }
+
+/* For a given address determine the sweep chunk boundaries that contains it, we expect this chunk to be assigned during reassignChunks.*/
+void
+MM_SweepHeapSectioning::getChunkBoundariesForAddr(MM_EnvironmentBase *env,  void* addrToCheck, void* &sweepChunkBaseBoundary, void* &sweepChunkTopBoundary)
```

Comment:
I'm thinking about simplifying (speeding up) this.
We could cache sweepChunkBase in MP and do the calculation there. We waste time here on figuring out the base for every  TLH allocation.

Reply:
I mean region base calcuation

Reply:
I shifted the code out of SweepHeapSectioning to the MP. In terms of caching the region, I don't think it can be done during INIT of MP (as we discussed), but we can fetch and cache it the first time we call this method. Another option is to set it when the region is created, i.e call MP to set the value after region init

Reply:
we can set/reset it when SATB is activated/deactivated
we would check if base is 0 or not instead of if SATB is active inside

Reply:
but not sure if env->_cycleState->_activeSubSpace is properly set, but if it is, then 
at the point we activate barrier in Collector we would do
env->_cycleState->_activeSubSpace->notifySATBActiveated()
then Subspace would do for its pools something like _memoryPool()->setTLHalignment(regionBase);
conversely, on barrier deactivate we would reset TLH alignment

Reply:
What do you think about an event?

For example MM_MemoryPoolLargeObjects registers on J9HOOK_MM_PRIVATE_GLOBAL_GC_INCREMENT_START to do LOA resizing during pre collect

``` C++
	(*mmPrivateHooks)->J9HookRegisterWithCallSite(mmPrivateHooks, J9HOOK_MM_PRIVATE_GLOBAL_GC_INCREMENT_START, reportGlobalGCIncrementStart, OMR_GET_CALLSITE(), (void*)this);

/**
 * Called at the start of a global collect.
 * We use this broadcast event to update target LOA ratio for this collect
 */
void
reportGlobalGCIncrementStart(J9HookInterface** hook, uintptr_t eventNum, void* eventData, void* userData)
```

Reply:
Generally, I find this pattern of using hooks to notify internal GC components among themselves overkill. Perhaps justified if there is more then one hook registered, but for just one, overkill.
I also try to avoid adding knowledge of higher level concepts (SATB is a thing in a Concurrent Global Collector) in lower level concepts (Memory Pool) here. MP should really not deal with things like SATB hook to register to, equally like it should not know about isSATBActive() check - it should just know that there are some kind of alignments imposed on TLHs from higher level entities . What entity imposed it, why, and when this alignment is active or not is not MP's business. It just need to perform that alignment if parameters are set and do nothing if they are not set.

That said, passing information when SATB is de/activated from Collector to MP directly via method invocations, while more correct in OO programming sense, does not seem to be trivial, since we might end up passing it through long chain of indirection Heap -> Space -> Subspace -> MP, introducing a lot of new methods, what may be even a more intrusive change.

So, i'm willing to accept the hook approach, since we already have it in MP class and there is also a possibility in future the same event could be used for some other entities to register to (like Scavenger). 

I believe only one event is needed, since we could pass a bool through user data if it's really activate or deactivate.